### PR TITLE
Quick Wins for Realm/React

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
Two small improvements:
* Allow `createRealmContext` to be called without an initial configuration
* Add a `fallback` property to `RealmProvider` that is shown while realm is opening

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: packages/realm-react/src/RealmProvider.tsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ import React, { useEffect, useState, useRef } from "react";
 import Realm from "realm";
 import { isEqual } from "lodash";
 
-type ProviderProps = Realm.Configuration;
+type ProviderProps = Realm.Configuration & {
+  fallback?: React.ComponentType<unknown> | React.ReactElement | null | undefined;
```
Comment (kraenhansen) [inline packages/realm-react/src/RealmProvider.tsx:24]: I suggest just taking the component and not a rendered element, this is what I've seen in other similar APIs.

```suggestion
  fallback?: React.ComponentType<unknown>;
```

File: packages/realm-react/src/RealmProvider.tsx [hunk 2]
```diff
@@ -109,7 +111,7 @@ export function createRealmProvider(
     }, [configVersion, realm, setRealm]);
 
     if (!realm) {
-      return null;
+      return <>{fallback}</>;
```
Comment (kraenhansen) [inline packages/realm-react/src/RealmProvider.tsx:114]:
```suggestion
      return <Fallback />;
```

File: packages/realm-react/src/__tests__/RealmProvider.test.tsx [hunk 3]
```diff
@@ -163,6 +175,24 @@ describe("RealmProvider", () => {
 
     expect(newSchemaNameContainer).toHaveTextContent("cat");
   });
+  it("initially renders a fallback, until realm exists", async () => {
+    const App = () => {
+      return (
+        <RealmProvider fallback={<View testID="fallbackContainer" />}>
```
Comment (kraenhansen) [inline packages/realm-react/src/__tests__/RealmProvider.test.tsx:181]:
```suggestion
        <RealmProvider fallback={() => <View testID="fallbackContainer" />}>
```

File: packages/realm-react/src/index.tsx [hunk 4]
```diff
@@ -123,11 +123,11 @@ type RealmContext = {
  * @param realmConfig - {@link Realm.Configuration} used to open the Realm
  * @returns An object containing a `RealmProvider` component, and `useRealm`, `useQuery` and `useObject` hooks
  */
-export const createRealmContext: (realmConfig: Realm.Configuration) => RealmContext = (
-  realmConfig: Realm.Configuration,
+export const createRealmContext: (realmConfig?: Realm.Configuration) => RealmContext = (
```
Comment (kraenhansen) [inline packages/realm-react/src/index.tsx:126]:
```suggestion
export const createRealmContext: (realmConfig: Realm.Configuration = {}) => RealmContext = (
```

File: packages/realm-react/src/index.tsx [hunk 5]
```diff
@@ -123,11 +123,11 @@ type RealmContext = {
...
+  realmConfig?: Realm.Configuration,
```
Comment (kraenhansen) [inline packages/realm-react/src/index.tsx:127]:
```suggestion
  realmConfig: Realm.Configuration = {},
```

File: packages/realm-react/CHANGELOG.md [hunk 6]
```diff
@@ -1,7 +1,8 @@
 x.x.x Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
-* None.
+* Allow `createRealmContext` to be called without an initial configuration
```
Comment (kneth) [inline packages/realm-react/CHANGELOG.md:4]:
```suggestion
* Allow `createRealmContext` to be called without an initial configuration.
```

**Review by kneth — APPROVED** [review]

File: packages/realm-react/src/RealmProvider.tsx (hunk 1 above)
Comment (takameyer) [reply packages/realm-react/src/RealmProvider.tsx:24]: I've modified the code to accept both elements and components

File: packages/realm-react/src/RealmProvider.tsx [hunk 7]
```diff
@@ -58,7 +60,7 @@ export function createRealmProvider(
    * For example, to override the `path` config value, use a prop named `path`,
    * e.g. `path="newPath.realm"`
    */
-  return ({ children, ...restProps }) => {
+  return ({ children, fallback, ...restProps }) => {
```
[the remark this answers is not in the document]
Comment (takameyer) [reply packages/realm-react/src/RealmProvider.tsx]: as did I

File: packages/realm-react/src/__tests__/RealmProvider.test.tsx [hunk 8]
```diff
@@ -163,6 +175,45 @@ describe("RealmProvider", () => {
 
     expect(newSchemaNameContainer).toHaveTextContent("cat");
   });
+  describe("initially renders a fallback, until realm exists", () => {
+    it("as a component", async () => {
+      const App = () => {
+        return (
+          <RealmProvider fallback={<View testID="fallbackContainer" />}>
```
Comment (kraenhansen) [inline packages/realm-react/src/__tests__/RealmProvider.test.tsx:182]: You need to pass a function (or class) to properly test this:

```suggestion
          <RealmProvider fallback={() => <View testID="fallbackContainer" />}>
```

**Review by kraenhansen — APPROVED** [review]

**Pull request merged**